### ISSUE TITLE
Unifying Item Prediction blocks (ItemsPrediction, ItemsPredictionWeightTying, ItemsPredictionSampled)

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/models#158]
The [NextItemPredictionTask](https://github.com/NVIDIA-Merlin/models/blob/main/merlin_models/tf/prediction/item_prediction.py#L601) uses three different item prediction blocks --`ItemsPrediction`, `ItemsPredictionWeightTying`, `ItemsPredictionSampled` -- with similar inputs and outputs.
This improvement issue is about unifying those three item prediction blocks into a single one, with arguments to set `weight_tying=True|False` and sampled item scoring (like in sampled softmax).
P.s. Those blocks should keep returning logits and not probs (like using softmax activation), as our losses functions will always expect logits.
